One Attestation API (#295)
LF-Decentralized-Trust-labs/LF-Decentralized-Trust-labs.github.io: labs/lfdt/one-attestation-api.md

---
layout: default
title: one-attestation-api

> parent: Labs

Reviewer: ```suggestion
parent: LFDT Labs
grand_parent: Active Labs
```

Author: Thank you Tracy!

---
# Lab Name
One Attestation API

# Short Description
This lab proposes to create a single API to handle remote attestations.
Attestations are a crucial mechanism to establish trust in hardware-based trusted execution environments (TEEs).
Today, multiple attestation mechanisms are available, like SGX-EPID (which is being phased out), SGX-DCAP, TDX-DCAP and SEV-attestation.
This proposal aims to create a single abstraction for the generation and verification of hardware-based attestations.
The abstraction can benefit, and avoid redundant implementations in, multiple projects (e.g., Fabric Private Chaincode, Private Data Objects) that use TEEs to enhance software integrity and confidentiality.

# Scope of Lab
The scope of the lab is to provide tools and support for hardware/software attestations.
Contributions from the community will help support additional attestation mechanisms and TEEs.

> Initially, we anticipate to integrate this tool in two Hyperledger projects: Fabric Private Chaincode and Private Data Objects.

Reviewer: ```suggestion
Initially, we anticipate to integrate this tool with: Fabric Private Chaincode and Private Data Objects.
```

Author: Thank you Tracy!
